drawer-menu/JS

sae-github opened the pull request:
# Drawer-menu

## [CodeSandbox](https://codesandbox.io/s/recursing-meninsky-fx1cn?file=/index.html)

### Specification Description
When you click on the 'hamburger icon', you will get the attribute value, and the style specified in the CSS will be hit.
Also, the class is given at that time, and the drawer menu will appear.

### Concerns and areas to focus on
- I'm not sure if I'm doing the PullRequest right...

sae-github committed:
update README

kenmori commented:
please upload js.file and the other.

kenmori commented:
you know, property and value has styled every two time
<img width="200" alt="スクリーンショット 2021-07-08 22 26 14" src="https://user-images.githubusercontent.com/9529003/124929484-8391e100-e03b-11eb-945b-50cc8fb01df7.png">

kenmori commented:
1. Do you need `bottom` property at `.p-hamburger `?? 

2. Did you try to use reset.css??

3. According to Material Design, duration expand and collapses is 200ms and 250ms. [see Exits and closing](https://material.io/design/motion/speed.html#duration).

4. use [will-change](https://developers.google.com/web/fundamentals/design-and-ux/animations/animations-and-performance) to transform. (After change it to animation speed less than 200ms)

5. navigation drawer  open left to right?? This is due to how the language is read.  [Material Design, Sheet Section](https://material.io/components/navigation-drawer#anatomy)

sae-github committed:
add: style.css index.js

sae-github commented:
> please upload js.file and the other.

I'm sorry.
I've uploaded the other files.

sae-github commented:
> please upload js.file and the other.

I'm sorry.
I've uploaded the other files.

sae-github commented:
> you know, property and value has styled every two time

I'm sorry.
I don't know the cause.
I checked the other demo sites and they all had the same problem...

sae-github committed:
Merge branch 'master' into develop

sae-github committed:
perf：index.html style.css index.js

sae-github committed:
Merge branch 'develop' of https://github.com/sae-github/drawer-menu into develop

sae-github commented:
**> * Do you need `bottom` property at `.p-hamburger `??**
It was an unneeded property.
It has been removed.

**> * Did you try to use reset.css??**
I did not use it this time.
Also, using it did not fix the double style.

**> * According to Material Design, duration expand and collapses is 200ms and 250ms.
Thank you.
I have looked into Material Design.
I also fixed the transition.

**> * use [will-change](https://developers.google.com/web/fundamentals/design-and-ux/animations/animations-and-performance) to transform. (After change it to animation speed less than 200ms)**
In index.js, we've specified that it will be added when the user hovers. It is also removed when the hover is removed.

**> * navigation drawer  open left to right?? This is due to how the language is read.  [Material Design, Sheet Section](https://material.io/components/navigation-drawer#anatomy)**
I'm really convinced.
I realized that changing it depending on the language will lead to ease of use for the user.

kenmori commented:
> I checked the other demo sites and they all had the same problem...

thank you for your checking. I see...

kenmori commented on src/index.js:
```diff
@@ -1,6 +1,13 @@
 const js_hamburger = document.getElementById("js-hamburger");
 const js_globalNav = document.getElementById("js-global-nav");
 
+js_hamburger.addEventListener("mouseover",function() {
+  js_globalNav.style.willChange = "transform";
+});
+js_hamburger.addEventListener("mouseout",function() {
+  js_globalNav.style.willChange = "";
+});
+
```
I think that this `js _` prefix is no need. because If you will be add it, all variables that exist in js file have to add it. Now, those variables clearly used to defined variables in js.

kenmori replied:
```suggestion
hamburger.addEventListener("mouseover",function() {
  globalNav.style.willChange = "transform";
});
hamburger.addEventListener("mouseout",function() {
  globalNav.style.willChange = "";
});

```

sae-github replied:
Thank you.
Fixed.
ed27567

kenmori commented:
`<!-- /.p-global-nav__list-->` 
I think Those comment is no need.

kenmori commented:
By the way,
I don't know that aria attribute should be use to control by js.

Do you have some reference this code is correct?
My perception is that you should avoid using aria in js, area must be written directly in HTML.

Setting to aria is ok, but `this.getAttribute("aria-expanded") === "false"` look like invalid in my understand.

your code here.
```js
hamburger.addEventListener("click", function () {
  globalNav.classList.toggle("is-active");
  if (this.getAttribute("aria-expanded") === "false") {
    this.setAttribute("aria-expanded", "true");
    globalNav.setAttribute("aria-hidden", "false");
  } else {
    this.setAttribute("aria-expanded", "false");
    globalNav.setAttribute("aria-hidden", "true");
  }
});
```

recommend is

```html
        <button type="button" data-isopen="false" id="js-hamburger" class="p-hamburger" aria-controls="js-glabal-menu" aria-expanded="false"
          area-label="メニューを開閉する">
```

in js

```js
hamburger.addEventListener("click", function () {
  globalNav.classList.toggle("is-active");
  if (e.currentTarget.dataset.isopen === "false") {
    this.setAttribute("aria-expanded", "true");
    globalNav.setAttribute("aria-hidden", "false");
  } else {
    this.dataset.isopen = "false";
    this.setAttribute("aria-expanded", "false");
    globalNav.setAttribute("aria-hidden", "true");
  }
});
```

or

just use `contains` 
```js
hamburger.addEventListener("click", function () {
  globalNav.classList.toggle("is-active");
  if (globalNav.classList.contains("is-active")) {
    this.setAttribute("aria-expanded", "true");
    globalNav.setAttribute("aria-hidden", "false");
  } else {
    this.setAttribute("aria-expanded", "false");
    globalNav.setAttribute("aria-hidden", "true");
  }
});
```

sae-github committed:
update:index.html index.js

sae-github committed:
Merge branch 'master' into develop

sae-github commented:
> `<!-- /.p-global-nav__list-->`
> I think Those comment is no need.

Thank you very much. I have made the correction.

sae-github commented:
Thank you very much. 

I did not understand and write the code.

Here is the site I was referring to.
https://shibajuku.net/make-hamburger-button/

I think it is based on this specification.
https://www.w3.org/TR/html-aria/

I was not sure if it was appropriate to control it with the aria attribute.
It was written in such a way that I could not properly explain the reason.

I modified it based on your code.

kenmori commented on index.html:
```diff
@@ -14,18 +14,17 @@
             <li class="p-global-nav__item"><a href="http://">メニュー4</a></li>
             <li class="p-global-nav__item"><a href="http://">メニュー5</a></li>
-          </ul><!-- /.p-global-nav__list-->
+          </ul>
         </nav><!-- /.p-header-nav -->
       </div><!-- /.l-inner -->
     </header><!-- /.l-header -->
```
nits

please remove those as same.
```js
   </nav><!-- /.p-header-nav -->
      </div><!-- /.l-inner -->
    </header><!-- /.l-header -->
    ```

sae-github replied:
Thank you.
Let me ask you a question.
Is it because this comment out is meaningless?
be66667

kenmori replied:
yes. [see](https://github.com/taka0209/js_lesson/pull/10#discussion_r659065149)

kenmori commented on src/index.js:
```diff
@@ -1,20 +1,22 @@
-js_hamburger.addEventListener("click",function () {
-  js_globalNav.classList.toggle("is-active");
-  if (this.getAttribute("aria-expanded") === "false") {
+hamburger.addEventListener("click",function () {
+  globalNav.classList.toggle("is-active");
+  if (globalNav.classList.contains("is-active")) {
+
```
IMO
this [code](https://www.w3.org/TR/wai-aria-practices-1.1/examples/accordion/js/accordion.js) is `wai-aria-practices-1.1/examples/accordion`

I found that it was using a class name.

```js
  accordion.addEventListener('click', function (event) {
    var target = event.target;

    if (target.classList.contains('Accordion-trigger')) {
      // Check if the current toggle is expanded.
``` 

I think the current code is more appropriate.

kenmori replied:
By the way, Are you going to apply aria and FLOCSS to all future code?

sae-github replied:
Thank you very much.
I'm learning so much.

sae-github replied:
> By the way, Are you going to apply aria and FLOCSS to all future code?

I will continue to use FLOCSS.
I will continue to use FLOCSS because I am researching it.
I also use FLOCSS in my work.

I don't intend to use the aria attribute for everything.
This time, I used it even though I didn't understand it very well...

kenmori replied:
> I don't intend to use the aria attribute for everything.
This time, I used it even though I didn't understand it very well...

ok. i understand.

> I will continue to use FLOCSS.
I will continue to use FLOCSS because I am researching it.
I also use FLOCSS in my work.

ok.
i recommended to use BEM for the more popular types,
Personally, BEM is not needed to learn the front-end.

sae-github committed:
update:index.html

sae-github committed:
Merge branch 'develop' of https://github.com/sae-github/drawer-menu into develop

kenmori commented:
Approve

sae-github commented:
Thank you very much.
